OpenRA.Game/Traits/World/Shroud.cs

```diff
@@ -383,13 +453,25 @@ public bool IsVisible(Actor a)
 
 		public bool IsTargetable(Actor a)
 		{
+			IEnumerable<CPos> vis = GetVisOrigins(a);
+			bool concealed = vis.All(c => generatedDisruptionCount[c] > 0 || generatedShroudCount[c] > 0);
```

Comment:
pchote: But I'm using occupied cells here to determine if a building is targetable. Should I use footprint here too? I have to use the same condition everywhere.